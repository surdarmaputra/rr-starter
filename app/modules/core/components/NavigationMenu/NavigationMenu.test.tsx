import { fireEvent, render, screen, within } from '@testing-library/react';
import { describe, expect, test } from 'vitest';

import {
  NavigationMenu,
  NavigationMenuContent,
  NavigationMenuItem,
  NavigationMenuLink,
  NavigationMenuList,
  NavigationMenuTrigger,
} from './NavigationMenu';

describe('NavigationMenu', () => {
  function renderComponent() {
    return render(
      <NavigationMenu data-testid="NavigationMenu-testid">
        <NavigationMenuList>
          <NavigationMenuItem data-testid="NavigationMenuItem1-testid">
            <NavigationMenuTrigger>Menu 1</NavigationMenuTrigger>
            <NavigationMenuContent data-testid="NavigationMenuContent1-testid">
              <ul className="p-4">
                <li>
                  <NavigationMenuLink>Item 1</NavigationMenuLink>
                </li>
                <li>
                  <NavigationMenuLink>Item 2</NavigationMenuLink>
                </li>
              </ul>
            </NavigationMenuContent>
          </NavigationMenuItem>
          <NavigationMenuItem data-testid="NavigationMenuItem2-testid">
            <NavigationMenuTrigger>Menu 2</NavigationMenuTrigger>
            <NavigationMenuContent data-testid="NavigationMenuContent2-testid">
              <ul className="p-4">
                <li>
                  <NavigationMenuLink>Item 3</NavigationMenuLink>
                </li>
                <li>
                  <NavigationMenuLink>Item 4</NavigationMenuLink>
                </li>
              </ul>
            </NavigationMenuContent>
          </NavigationMenuItem>
          <NavigationMenuItem data-testid="NavigationMenuItem3-testid">
            <NavigationMenuLink>Menu 3</NavigationMenuLink>
          </NavigationMenuItem>
        </NavigationMenuList>
      </NavigationMenu>,
    );
  }

  test('renders with data-testid', async () => {
    renderComponent();
    const navigationMenu = await screen.findByTestId('NavigationMenu-testid');

    const navigationMenuItem1 = await within(navigationMenu).findByTestId(
      'NavigationMenuItem1-testid',
    );
    expect(
      screen.queryByTestId('NavigationMenuContent1-testid'),
    ).not.toBeInTheDocument();
    fireEvent.click(
      within(navigationMenuItem1).getByRole('button', { name: /Menu 1/ }),
    );
    const menuList1 = await screen.findByTestId(
      'NavigationMenuContent1-testid',
    );
    const menuList1Items = within(menuList1).getAllByRole('listitem');
    expect(menuList1Items).toHaveLength(2);
    expect(menuList1Items[0]).toHaveTextContent('Item 1');
    expect(menuList1Items[1]).toHaveTextContent('Item 2');

    const navigationMenuItem2 = await within(navigationMenu).findByTestId(
      'NavigationMenuItem2-testid',
    );
    expect(
      screen.queryByTestId('NavigationMenuContent2-testid'),
    ).not.toBeInTheDocument();
    fireEvent.click(
      within(navigationMenuItem2).getByRole('button', { name: /Menu 2/ }),
    );
    const menuList2 = await screen.findByTestId(
      'NavigationMenuContent2-testid',
    );
    const menuList2Items = within(menuList2).getAllByRole('listitem');
    expect(menuList2Items).toHaveLength(2);
    expect(menuList2Items[0]).toHaveTextContent('Item 3');
    expect(menuList2Items[1]).toHaveTextContent('Item 4');
  });

  test('renders snapshot', async () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
