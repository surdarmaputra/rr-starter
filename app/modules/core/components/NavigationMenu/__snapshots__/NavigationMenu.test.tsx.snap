// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavigationMenu > renders snapshot 1`] = `
<div>
  <nav
    aria-label="Main"
    class="relative z-10 flex max-w-max flex-1 items-center justify-center"
    data-orientation="horizontal"
    data-testid="NavigationMenu-testid"
    dir="ltr"
  >
    <div
      style="position: relative;"
    >
      <ul
        class="group flex flex-1 list-none items-center justify-center space-x-1"
        data-orientation="horizontal"
        dir="ltr"
      >
        <li
          data-testid="NavigationMenuItem1-testid"
        >
          <button
            aria-controls="radix-:r4:-content-radix-:r5:"
            aria-expanded="false"
            class="group inline-flex h-10 w-max items-center justify-center rounded-md bg-white px-4 py-2 text-sm font-semibold transition-colors hover:bg-slate-100 hover:text-slate-900 focus:bg-slate-100 focus:text-slate-900 focus:outline-none disabled:pointer-events-none disabled:opacity-50 data-[active]:bg-slate-100/50 data-[state=open]:bg-slate-100/50 dark:bg-slate-950 dark:hover:bg-slate-800 dark:hover:text-slate-50 dark:focus:bg-slate-800 dark:focus:text-slate-50 dark:data-[active]:bg-slate-800/50 dark:data-[state=open]:bg-slate-800/50 group"
            data-radix-collection-item=""
            data-state="closed"
            id="radix-:r4:-trigger-radix-:r5:"
          >
            Menu 1
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-down relative top-[1px] ml-1 h-3 w-3 transition duration-200 group-data-[state=open]:rotate-180"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6 9 6 6 6-6"
              />
            </svg>
          </button>
        </li>
        <li
          data-testid="NavigationMenuItem2-testid"
        >
          <button
            aria-controls="radix-:r4:-content-radix-:r6:"
            aria-expanded="false"
            class="group inline-flex h-10 w-max items-center justify-center rounded-md bg-white px-4 py-2 text-sm font-semibold transition-colors hover:bg-slate-100 hover:text-slate-900 focus:bg-slate-100 focus:text-slate-900 focus:outline-none disabled:pointer-events-none disabled:opacity-50 data-[active]:bg-slate-100/50 data-[state=open]:bg-slate-100/50 dark:bg-slate-950 dark:hover:bg-slate-800 dark:hover:text-slate-50 dark:focus:bg-slate-800 dark:focus:text-slate-50 dark:data-[active]:bg-slate-800/50 dark:data-[state=open]:bg-slate-800/50 group"
            data-radix-collection-item=""
            data-state="closed"
            id="radix-:r4:-trigger-radix-:r6:"
          >
            Menu 2
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-down relative top-[1px] ml-1 h-3 w-3 transition duration-200 group-data-[state=open]:rotate-180"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6 9 6 6 6-6"
              />
            </svg>
          </button>
        </li>
        <li
          data-testid="NavigationMenuItem3-testid"
        >
          <a
            data-radix-collection-item=""
          >
            Menu 3
          </a>
        </li>
      </ul>
    </div>
    <div
      class="absolute left-0 top-full flex justify-center"
    />
  </nav>
</div>
`;
